### PR TITLE
Task/4

### PR DESCRIPTION
**Pull request by Sassoricun**

### CODE REVIEW

File: src/app/components/color-panel/color-panel.component.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  styleUrls: ['./color-panel.component.scss']
+})
+export class ColorPanelComponent implements OnInit {
+  @Output() emitColor: EventEmitter<string> = new EventEmitter();
+
+  colorPickerOpen: boolean = false;
+
+  colorsData = Object.values(colors)
+
+
```
Comment (RomanYezerskyi) [inline src/app/components/color-panel/color-panel.component.ts:24]: extra lines

File: src/app/components/dashboard-item/dashboard-item.component.html [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<div class="top">
+  <div>{{item.text}}</div>
+  <mat-icon class="icon-btn" color="primary" (click)="onCardDelete(item.id)">delete</mat-icon>
+</div>
+
```
Comment (RomanYezerskyi) [inline src/app/components/dashboard-item/dashboard-item.component.html:5]: extra line

File: src/app/components/dashboard-item/dashboard-item.component.scss [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  cursor: pointer;
+}
+
+.down {
+  display: flex;
+  align-items: center;
+  justify-content: flex-start;
+  height: 100%;
+}
+
```
Comment (RomanYezerskyi) [inline src/app/components/dashboard-item/dashboard-item.component.scss:23]: extra line

File: src/app/components/dashboard-item/dashboard-item.component.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+
+@Component({
+  selector: 'app-dashboard-item',
+  templateUrl: './dashboard-item.component.html',
+  styleUrls: ['./dashboard-item.component.scss']
+})
+export class DashboardItemComponent implements OnInit {
+
+  @Input() item;
```
Comment (RomanYezerskyi) [inline src/app/components/dashboard-item/dashboard-item.component.ts:10]: what the type of field 'item' ?

File: src/app/components/dashboard/dashboard.component.html [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        <div class="column-header__btns">
+          <app-dialog (emitText)="onAddCard($event, column.id)"
+            [question]="'Add new card in ' + column.title + ' column'">
+            <span class="material-symbols-outlined">add</span>
+          </app-dialog>
+          <mat-icon class="btn-delete" color="white" (click)="onDeleteColumn(column.id)">delete</mat-icon>
+          <app-color-panel (emitColor)="onColorChange($event, column.id)"></app-color-panel>
+        </div>
+      </div>
+
```
Comment (RomanYezerskyi) [inline src/app/components/dashboard/dashboard.component.html:15]: extra line

File: src/app/components/dashboard/dashboard.component.scss [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  color: white;
+  align-items: center;
+  justify-content: space-between;
+  box-sizing: border-box;
+  cursor: move;
+  background: #cfd8dc;
+  font-size: 14px;
+}
+
+.cdk-drag-preview {
```
Comment (RomanYezerskyi) [inline src/app/components/dashboard/dashboard.component.scss:46]: What is cdk-drag ? Also, does the license allow us to change something?

File: src/app/components/dashboard/dashboard.component.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  selector: 'app-dashboard',
+  templateUrl: './dashboard.component.html',
+  styleUrls: ['./dashboard.component.scss'],
+})
+export class DashboardComponent implements OnInit {
+
+  constructor(public dashboardService: DashboardService) { }
+
+  ngOnInit(): void {
+    console.log('BOARD - INIT');
```
Comment (RomanYezerskyi) [inline src/app/components/dashboard/dashboard.component.ts:19]: remove

File: src/app/components/dialog-body/dialog-body.component.html [hunk 8]
```diff
@@ -0,0 +1,11 @@
+<div mat-dialog-content>
+  <h3>{{data.question}}</h3>
+  <mat-form-field style="width: 100%;">
```
Comment (RomanYezerskyi) [inline src/app/components/dialog-body/dialog-body.component.html:3]: styles should be in css file

File: src/app/components/dialog-body/dialog-body.component.html [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+    <mat-label>Add Title</mat-label>
+    <input matInput [(ngModel)]="data.text">
+  </mat-form-field>
+</div>
+<div mat-dialog-actions style="padding-left: 23px;">
```
Comment (RomanYezerskyi) [inline src/app/components/dialog-body/dialog-body.component.html:8]: styles should be in css file

File: src/app/components/dialog-body/dialog-body.component.ts [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+  ) { }
+
+  onNoClick(): void {
+    this.dialogRef.close();
+  }
+
+  ngOnInit(): void {
+
+  }
+
```
Comment (RomanYezerskyi) [inline src/app/components/dialog-body/dialog-body.component.ts:23]: extra line

File: src/app/services/dashboard.service.ts [hunk 11]
```diff
@@ -0,0 +1,188 @@
...
+            },
+            {
+              id: 3,
+              text: 'Some comment'
+            }
+          ]
+        }
+      ]
+    }
+  ]
```
Comment (RomanYezerskyi) [inline src/app/services/dashboard.service.ts:64]: this data should be in helper file

File: tsconfig.json [hunk 12]
```diff
@@ -5,7 +5,7 @@
     "baseUrl": "./",
     "outDir": "./dist/out-tsc",
     "forceConsistentCasingInFileNames": true,
-    "strict": true,
+    "strict": false,
```
Comment (RomanYezerskyi) [inline tsconfig.json:8]: why ?

File: src/app/components/dashboard-item/dashboard-item.component.scss [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+.icon {
+  margin-left: 8px;
+  margin-bottom: 15px;
+  cursor: pointer;
+}
+
+.num {
+  transform: translateX(5px);
+  transform: translateY(5px);
```
[the remark this answers is not in the document]
Comment (Sassoricun) [reply src/app/components/dashboard-item/dashboard-item.component.scss:33]: delete, not important

File: src/app/components/dashboard-item/dashboard-item.component.ts [hunk 14]
```diff
@@ -7,7 +7,7 @@ import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
 })
 export class DashboardItemComponent implements OnInit {
 
-  @Input() item;
+  @Input() item: any;
```
Comment (RomanYezerskyi) [inline src/app/components/dashboard-item/dashboard-item.component.ts:10]: any is very bad type ))
Maybe we can use something else ?

File: src/app/components/dialog/dialog.component.ts [hunk 15]
```diff
@@ -9,7 +9,7 @@ import { DialogBodyComponent } from '../dialog-body/dialog-body.component';
 })
 export class DialogComponent implements OnInit {
   @Output() emitText: EventEmitter<any> = new EventEmitter()
-  @Input() question: string;
+  @Input() question: any;
```
Comment (RomanYezerskyi) [inline src/app/components/dialog/dialog.component.ts:12]: any ?

File: src/app/services/dashboard.service.ts [hunk 16]
```diff
@@ -166,12 +112,12 @@ export class DashboardService {
     this.dashboard$.next([...this.dasboard]);
   }
 
-  deleteComment(columnId, itemId, commentId) {
-    this.dasboard = this.dasboard.map((column) => {
+  deleteComment(columnId: number, itemId: number, commentId: number) {
+    this.dasboard = this.dasboard.map((column: any) => {
```
Comment (RomanYezerskyi) [inline src/app/services/dashboard.service.ts:116]: any ?

**Review by RomanYezerskyi — COMMENTED** [review]
We should avoid using 'any' type

**Pull request merged**